File: java/atividade4/Temperatura.java
/**
 *  Este material foi criado com fins de estudos e aprendizagem,
 * com o objetivo de divulgar e demonstrar os meus códigos e a
 * minha evolução na área.
 * 
 * Autor: Josue Lopes
 * 
 * Temperatura.java
 */
public class Temperatura {

    /**
     * Recebe a temperatura na escala de Kelvin
     * e converte ela para Celsius.
     * 
     * @param kelvin temperatura em Kelvin.
     * @return temperatura em Celsius.
     */
    public float kelvinToCelsius(float kelvin){
        return (kelvin - 273.15f);
    }

    /**
     * Recebe a temperatura na escala de Kelvin
     * e converte ela para Fahrenheit.
     * 
     * @param kelvin temperatura em Kelvin.
     * @return temperatura em Fahrenheit.
     */
    public float kelvinToFahrenheit(float kelvin){
        return ((kelvin - 273.15f) * 9f/5f + 32f);
    }

    /**
     * Recebe a temperatura na escala de Celsius
     * e converte ela para Kelvin.
     * 
     * @param celsius temperatura em Celsius.
     * @return temperatura em Kelvin.
     */
    public float celsiusToKelvin(float celsius){
        return celsius + 273.15f;
    }

    /**
     * Recebe a temperatura na escala de Celsius
     * e converte ela para Fahrenheit.
     * 
     * @param celsius temperatura em Celsius.
     * @return temperatura em Fahrenheit.
     */
    public float celsiusToFahrenheit(float celsius){
        return ((celsius*9f/5f) + 32f);
    }

    /**
     * Recebe a temperatura na escala de Fahrenheit
     * e converte ela para Kelvin.
     * 
     * @param fahrenheit temperatura em Fahrenheit.
     * @return temperatura em Kelvin.
     */
    public float fahrenheitToKelvin(float fahrenheit) {
        return ((fahrenheit - 32f) * 5f/9f + 273.15f);
    }

    /**
     * Recebe a temperatura na escala de Fahrenheit
     * e converte ela para Celsius.
     * 
     * @param fahrenheit temperatura em Fahrenheit.
     * @return temperatura em Celsius.
     */
    public float fahrenheitToCelsius(float fahrenheit){
        return ((fahrenheit - 32f) * 5f/9f);
    }

    /**
     * Recebe a temperatura e a escala em que a temperatura está
     * com os valores a temperatura é convertida na outras escalas
     * e retorna uma string com todos os dados.
     * 
     * @param escala seleciona qual escala é a temperatura 1 = Celsius, 2 = Fahrenheit e 3 = Kelvin.
     * @param temperatura recebe a temperatura.
     * @return strign com todas as conversões.
     */
    public String convert(int escala, float temperatura){
        StringBuilder txt = new StringBuilder();
        float celsius = 0, fahrenheit = 0, kelvin = 0;
        if (escala == 1 || escala == 2 || escala == 3){
            if (escala == 1){
                celsius = temperatura;
                fahrenheit = celsiusToFahrenheit(celsius);
                kelvin = celsiusToKelvin(celsius);
            } else if (escala == 2) {
                fahrenheit = temperatura;
                celsius = fahrenheitToCelsius(fahrenheit);
                kelvin = fahrenheitToKelvin(fahrenheit);
            } else if (escala == 3) {
                kelvin = temperatura;
                celsius = kelvinToCelsius(kelvin);
                fahrenheit = kelvinToFahrenheit(kelvin);
            }
            txt.append("\nCelsius " + celsius + "°C\n");
            txt.append("Fahrenheit " + fahrenheit + "F\n");
            txt.append("Kelvin " + kelvin + "K\n");
            return(txt.toString());
        } else {
            return"";
        }
    }
}

File: python/atividade4/temperatura.py
# coding=utf-8
'''
  Este material foi criado com fins de estudos e aprendizagem,
 com o objetivo de divulgar e demonstrar os meus códigos e a
 minha evolução na área.
 
 Autor: Josue Lopes

 temperatura.py
'''
class Temperatura:
    # Verifica se a escala passada existe.
    def checaEscala(self, escala):
        return checaEscalaCelsius(escala) or checaEscalaFahrenheit(escala) or checaEscalaKelvin(escala)

    # Recebe a escala e converte para a temperatura desejada.
    def converteTemperatura(self, escala, temperatura):
        print("------------------------------------------")
        if (checaEscalaCelsius(escala)):
            print("Temperatura em Celsius: " + str(temperatura))
            print("Temperatura em Fahrenheit: " + str(celsiusToFahrenheit(temperatura)))
            print("Temperatura em Kelvin: " + str(celsiusToKelvin(temperatura)))
        elif (checaEscalaFahrenheit(escala)):
            print("Temperatura em Fahrenheit: " + str(temperatura))
            print("Temperatura em Celsius: " + str(fahrenheitToCelsius(temperatura)))
            print("Temperatura em Kelvin: " + str(fahrenheitToKelvin(temperatura)))
        elif (checaEscalaKelvin(escala)):
            print("Temperatura em Kelvin: " + str(temperatura))
            print("Temperatura em Celsius: " + str(kelvinToCelsius(temperatura)))
            print("Temperatura em Fahrenheit: " + str(kelvinToFahrenheit(temperatura)))
        else:
            print("ERRO : Escala incorreta!")
        print("------------------------------------------")

#END CLASS

# Converte a temperatura na escala Celsius para Kelvin
def celsiusToKelvin(celsius):
    return (celsius + 273.15)

# Converte a temperatura na escala Celsius para Fahrenheit.
def celsiusToFahrenheit(celsius):
    return ((celsius*9/5) + 32)

# Converte a temperatura na escala Fahrenheit para a escala Celsius.
def fahrenheitToCelsius(fahrenheit):
    return ((fahrenheit - 32) * 5/9)

# Converte a temperatura na escala Fahrenheit para a escala Kelvin.
def fahrenheitToKelvin(fahrenheit):
    return ((fahrenheit - 32) * 5/9 + 273.15)

# Converte a temperatura na escala Kelvin para Celsius.
def kelvinToCelsius(kelvin):
    return (kelvin - 273.15)

# Converte a temperatura na escala Kelvin para Fahrenheit.
def kelvinToFahrenheit(kelvin):
    return ((kelvin - 273.15) * 9/5 + 32)

# Verifica se é a escala Celsius.
def checaEscalaCelsius(escala):
    return (escala == "1" or escala.lower() == "celsius" or escala.lower() == "1-celsius")

# Verifica se é a escala Fahrenheit
def checaEscalaFahrenheit(escala):
    return (escala == "2" or escala.lower() == "fahrenheit" or escala.lower() == "2-fahrenheit")

# Verifica se é a escala Kelvin
def checaEscalaKelvin(escala):
    return (escala == "3" or escala.lower() == "kelvin" or escala.lower() == "3-kelvin")